lib/PhasarLLVM/ControlFlow/Resolver/OTFResolver.cpp

```diff
@@ -105,29 +105,34 @@ auto OTFResolver::resolveVirtualCall(const llvm::CallBase *CallSite)
 
   const llvm::Value *Receiver = CallSite->getArgOperand(0);
 
-  // Use points-to information to resolve the indirect call
-  auto AllocSites = PT.getReachableAllocationSites(Receiver);
-  auto PossibleAllocatedTypes = getReachableTypes(*AllocSites);
-
-  // Now we must check if we have found some allocated struct types
-  set<const llvm::StructType *> PossibleTypes;
-  for (const auto *Type : PossibleAllocatedTypes) {
-    if (const auto *StructType =
-            llvm::dyn_cast<llvm::StructType>(stripPointer(Type))) {
-      PossibleTypes.insert(StructType);
-    }
-  }
+  if (CallSite->getCalledOperand() &&
+      CallSite->getCalledOperand()->getType()->isPointerTy()) {
+    if (const auto *FTy = llvm::dyn_cast<llvm::FunctionType>(
+            CallSite->getCalledOperand()->getType()->getPointerElementType())) {
 
-  for (const auto *PossibleTypeStruct : PossibleTypes) {
-    const auto *Target =
-        getNonPureVirtualVFTEntry(PossibleTypeStruct, VtableIndex, CallSite);
-    if (Target) {
-      PossibleCallTargets.insert(Target);
+      auto PTS = PT.getPointsToSet(CallSite->getCalledOperand(), CallSite);
+      for (const auto *P : *PTS) {
+        if (auto *PGV = llvm::dyn_cast<llvm::GlobalVariable>(P)) {
+          if (PGV->hasName() && PGV->getName().startswith("_ZTV") &&
+              PGV->hasInitializer()) {
+            if (auto *PCS = llvm::dyn_cast<llvm::ConstantStruct>(
+                    PGV->getInitializer())) {
+              auto VFs = LLVMVFTable::getVFVectorFromIRVTable(PCS);
+              if (VtableIndex < 0 || VtableIndex >= VFs.size()) {
```

Comment:
Is `VtableIndex < 0` an actual "wanted" condition?
If not, maybe we should put this check as an assert.

Reply:
I am not sure about this, depends on code that is generated.

Reply:
What code would need/use a negative index into a VTable? This would be highly undefined behavior.

Reply:
Actually, the current implementation returns negative indices as error codes, so we should check for that

Reply:
Then, this sounds for a good place to add an assert to not run into trouble later.
We can still keep the check, so not to crash during a release build but I would assume we want to know if an error code is reaching this location in the code.

Reply:
Actually it was really pain for me to see these negative error codes, terrible! I'm gonna do this right with std::optional now.